src/ed25519.rs

```diff
@@ -758,6 +843,63 @@ impl PublicKey {
 
         (signature.r.as_bytes()).ct_eq(r.compress().as_bytes()).unwrap_u8() == 1
     }
+
+    /// Verify a `signature` on a `prehashed_message` using the Ed25519ph algorithm.
+    ///
+    /// # Inputs
+    ///
+    /// * `prehashed_message` is an instantiated hash digest with 512-bits of
+    ///   output which has had the message to be signed previously fed into its
+    ///   state.
+    /// * `context` is an optional context string, up to 255 bytes inclusive,
+    ///   which may be used to provide additional domain separation.  If not
+    ///   set, this will default to an empty string.
+    /// * `signature` is a purported Ed25519ph [`Signature`] on the `prehashed_message`.
+    ///
+    /// # Returns
+    ///
+    /// Returns `true` if the `signature` was a valid signature created by this
+    /// `Keypair` on the `prehashed_message`.
+    ///
+    /// [rfc8032]: https://tools.ietf.org/html/rfc8032#section-5.1
+    pub fn verify_prehashed<D>(&self,
+                               prehashed_message: D,
+                               context: Option<&[u8]>,
+                               signature: &Signature) -> bool
+        where D: Digest<OutputSize = U64> + Default
+    {
+        let mut h: D = D::default();
+        let mut hash: [u8; 64] = [0u8; 64];
+
+        let mut a: EdwardsPoint = match self.decompress() {
+            Some(x) => x,
+            None    => return false,
+        };
+        a = -(&a);
+
+        let ctx: &[u8] = match context {
+            Some(x) => x,
+            None    => b"",  // By default, the context is an empty string.
+        };
+        debug_assert!(ctx.len() <= 255, "The context must not be longer than 255 octets.");
+
+        let ctx_len: u8 = ctx.len() as u8;
+
+        h.input(b"SigEd25519 no Ed25519 collisions");
+        h.input(&[1]); // Ed25519ph
+        h.input(&[ctx_len]);
+        h.input(ctx);
+        h.input(signature.r.as_bytes());
+        h.input(self.as_bytes());
+        h.input(prehashed_message.fixed_result().as_slice());
+        hash.copy_from_slice(h.fixed_result().as_slice());
+
+        let digest_reduced: Scalar = Scalar::from_bytes_mod_order_wide(&hash);
+        let r: EdwardsPoint = EdwardsPoint::vartime_double_scalar_mul_basepoint(&digest_reduced,
+                                                                                &a, &signature.s);
+
+        (signature.r.as_bytes()).ct_eq(r.compress().as_bytes()).unwrap_u8() == 1
```

Comment:
This can just be `r.compress() == signature.r` (no CT needed)

Reply:
Yep! I was going to change it to match your changes from #30 after the latter gets merged.